### PR TITLE
Add TLS and Token support

### PR DESCRIPTION
**Pull request by liuhaoyang**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [ ] Bug fix
- [x] New feature provided
- [ ] Improve performance

- Related issues
https://github.com/OpenSkywalking/skywalking-netcore/issues/22
https://github.com/OpenSkywalking/skywalking-netcore/issues/21
___
### New feature or improvement
- Describe the details and related test reports.

Add TLS support.

### CODE REVIEW

**Comment by wu-sheng** [comment]
Good feature.

File: src/SkyWalking.AspNet/SkyWalkingOptions.cs [hunk 1]
```diff
@@ -55,5 +55,23 @@ public class SkyWalkingOptions
         /// Zero or minus value means no limit.
         /// </summary>
         public int PendingSegmentsLimit { get; set; } = 300000;
+        
+        /// <summary>
+        /// Set your own token to active auth
+        /// </summary>
+        public string Authentication
```
Comment (yuleyule66) [inline src/SkyWalking.AspNet/SkyWalkingOptions.cs]: Is it better rename `Authentication` to `AuthenticationToken` ?

Comment (yuleyule66) [inline src/SkyWalking.AspNet/SkyWalkingOptions.cs]: Is it better rename `Authentication` to `AuthenticationToken` ?

File: src/SkyWalking.Abstractions/Config/CollectorConfig.cs [hunk 2]
```diff
@@ -35,5 +35,9 @@ public class CollectorConfig
         /// Collector cluster：DirectServers="10.2.45.126:11800,10.2.45.127:11800"
         /// </summary>
         public static string DirectServers { get; set; }
+        
+        public static string Authentication { get; set; }
```
Comment (yuleyule66) [inline src/SkyWalking.Abstractions/Config/CollectorConfig.cs]: Is it better rename `Authentication` to `AuthenticationToken` ?

File: src/SkyWalking.AspNet/SkyWalkingOptions.cs (hunk 1 above)
Comment (liuhaoyang) [reply src/SkyWalking.AspNet/SkyWalkingOptions.cs]: Refer to the [Set Token](https://github.com/apache/incubator-skywalking/blob/master/docs/en/Token-auth.md#set-token). I prefer to use the same config name.

**Pull request merged**